layers/meta-balena-raspberrypi/recipes-kernel/linux/linux-balena-bootloader_6.1.bb

```diff
@@ -0,0 +1,50 @@
+LINUX_VERSION ?= "6.1.64"
+LINUX_RPI_BRANCH ?= "rpi-6.1.y"
+LINUX_RPI_KMETA_BRANCH ?= "yocto-6.1"
+
+SRCREV_machine = "dad5d57939cfae7af363e7c9862b59d33d96794b"
+SRCREV_meta = "f845a7f37d7114230d6609e2bd630070f2f6cd9b"
+
+KMETA = "kernel-meta"
+
+SRC_URI = " \
+          git://github.com/raspberrypi/linux.git;name=machine;branch=${LINUX_RPI_BRANCH};protocol=https \
+          git://git.yoctoproject.org/yocto-kernel-cache;type=kmeta;name=meta;branch=${LINUX_RPI_KMETA_BRANCH};destsuffix=${KMETA} \
+"
+SRC_URI:remove = "file://initramfs-image-bundle.cfg"
+SRC_URI:remove = "file://vc4graphics.cfg"
+
+require recipes-kernel/linux/linux-raspberrypi.inc
+
+inherit balena-bootloader
+
+BALENA_DEFCONFIG_NAME = "${KBUILD_DEFCONFIG}"
+
+BALENA_CONFIGS:append = " \
+    rpisense \
+    "
+
+BALENA_CONFIGS[rpisense] = " \
+    CONFIG_MFD_RPISENSE_CORE=n \
+    CONFIG_FB_RPISENSE=n \
+    "
+
+do_deploy:append () {
+    BOOTENV_FILE="${DEPLOYDIR}/${KERNEL_PACKAGE_NAME}/bootenv"
+    grub-editenv "${BOOTENV_FILE}" create
+    grub-editenv "${BOOTENV_FILE}" set "resin_root_part=A"
+    grub-editenv "${BOOTENV_FILE}" set "bootcount=0"
+    grub-editenv "${BOOTENV_FILE}" set "upgrade_available=0"
+}
+
+do_deploy[depends] += " grub-native:do_populate_sysroot"
+
+KERNEL_DTC_FLAGS += "-@ -H epapr"
+
+INITRAMFS_IMAGE = "balena-image-bootloader-initramfs"
+
+KERNEL_PACKAGE_NAME = "balena-bootloader"
+
+KERNEL_DEVICETREE = "${RPI_KERNEL_DEVICETREE}"
+
+PROVIDES = "virtual/balena-bootloader"
```

Comment:
why not "virtual/bootloader"?

Reply:
This is used to pull balena bootloader into the image without using the specific package name:
https://github.com/balena-os/balena-raspberrypi/pull/1091/files#diff-abca8cd7f0af2e032dfdeec9e689532b73da44e8b689133f6d02617cec7fb5f9R79

But I am not sure whether it is particularly useful since the package name will be per-device-repo anyway. That is the only place where it is used. I think it is `virtual/balena-bootloader` because it was `virtual/linux-kboot` in the old dev branches and I just changed `linux-kboot` to `balena-bootloader` everywhere.

I have no strong opinion either way, do you think we should provide `virtual/bootloader` here and pull `linux-balena-bootloader` into the package?

Reply:
@mtoman I think it's cleaner to use `virtual/bootloader` as currently that will build u-boot. Not introducing a second virtual target simplifies things.